src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataMiamElementsMapper.java

```diff
@@ -0,0 +1,172 @@
+    }
+
+    private static List<MiamChildProtectionConcernChecklistEnum>
+        buildMiamChildProtectionConcernList(C100RebuildMiamElements c100RebuildMiamElements) {
+        List<String> childProtectionEvidences = nonNull(c100RebuildMiamElements.getMiamChildProtectionEvidence())
+                ? List.of(c100RebuildMiamElements.getMiamChildProtectionEvidence()) : Collections.emptyList();
+
+        if (childProtectionEvidences.isEmpty() || childProtectionEvidences.contains("none")) {
+            return null;
+        }
+        return childProtectionEvidences.stream().map(value -> ChildProtectionMapperEnum.valueOf(value).getValue())
+                .collect(Collectors.toList());
+    }
+
+    private static List<MiamDomesticViolenceChecklistEnum> buildMiamDomesticViolenceChecklist(C100RebuildMiamElements
+                                                                                                      c100RebuildMiamElements) {
+        List<String> domesticAbuses = nonNull(c100RebuildMiamElements.getMiamDomesticAbuse())
+                ? List.of(c100RebuildMiamElements.getMiamDomesticAbuse()) : Collections.emptyList();
+
+        if (domesticAbuses.isEmpty() || domesticAbuses.contains("none")) {
+            return null;
+        }
+
+        String[] domesticAbuse1 = c100RebuildMiamElements.getMiamDomesticAbuseInvolvementSubfields();
+        String[] domesticAbuse2 = c100RebuildMiamElements.getMiamDomesticAbuseCourtInvolvementSubfields();
+        String[] domesticAbuse3 = c100RebuildMiamElements.getMiamDomesticAbuseLetterOfBeingVictimSubfields();
+        String[] domesticAbuse4 = c100RebuildMiamElements.getMiamDomesticAbuseLetterFromAuthoritySubfields();
+        String[] domesticAbuse5 = c100RebuildMiamElements.getMiamDomesticAbuseLetterFromSupportServiceSubfields();
+
+        List<String> domesticAbuse1List = nonNull(domesticAbuse1) ? List.of(domesticAbuse1) : Collections.emptyList();
+        List<String> domesticAbuse2List = nonNull(domesticAbuse2) ? List.of(domesticAbuse2) : Collections.emptyList();
+        List<String> domesticAbuse3List = nonNull(domesticAbuse3) ? List.of(domesticAbuse3) : Collections.emptyList();
+        List<String> domesticAbuse4List = nonNull(domesticAbuse4) ? List.of(domesticAbuse4) : Collections.emptyList();
+        List<String> domesticAbuse5List = nonNull(domesticAbuse5) ? List.of(domesticAbuse5) : Collections.emptyList();
+
+        List<String> domesticAbusesList = Stream.of(domesticAbuse1List, domesticAbuse2List, domesticAbuse3List,
+                        domesticAbuse4List, domesticAbuse5List)
+                .flatMap(Collection::stream).collect(Collectors.toList());
+
+        domesticAbusesList.removeIf(Strings::isNullOrEmpty);
+
+        List<String> domesticAbuseHeads = List.of(c100RebuildMiamElements.getMiamDomesticAbuse());
+
+        if (domesticAbuseHeads.contains("c100RebuildMiamElements")) {
```

Comment:
@rahulsanjava19 What's the significance of this ? Why are we adding header one by one ?
Headers will be added line139 right ?
session data.
`"miam_domesticAbuse": [
"policeInvolvement",
"courtInvolvement",
"letterOfBeingVictim",
"letterFromAuthority",
"letterFromSupportService",
"ILRDuetoDomesticAbuse",
"financiallyAbuse"
],`

Reply:
@rahulsanjava19 as discussed please fix this & push this as part of applicant mapping PR.

Reply:
done